### PR TITLE
Add Plugin extension

### PR DESCRIPTION
**Pull request by danigm**
This new extension point will allow to extend the kolibri app with external flatpaks that just provides new extensions as python modules without the need to update the base app.

This is an example plugin manifest: https://github.com/danigm/org.learningequality.Kolibri.Plugin.Explore/blob/master/org.learningequality.Kolibri.Plugin.Explore.json

This is a rebased version of this one #19

### CODE REVIEW

**Comment by danigm** [comment]
@jamalex can you take a look to this PR? It's needed for the endless key to be able to send the custom presentation explore plugin: https://github.com/flathub/flathub/pull/2103

**Comment by jamalex** [comment]
@danigm I've tagged you now on a discussion thread in Slack about this. Thanks!

File: org.learningequality.Kolibri.yaml [hunk 1]
```diff
@@ -15,13 +15,19 @@ finish-args:
...
   - --env=KOLIBRI_HOME=~/.var/app/org.learningequality.Kolibri/data/kolibri
+  - --env=PYTHONPATH=/app/kolibri-plugins/lib/python
 
 add-extensions:
   org.learningequality.Kolibri.Content:
     version: '1.0'
     directory: share/kolibri-content
     subdirectories: true
     no-autodownload: true
+  org.learningequality.Kolibri.Plugin:
```
Comment (wjt) [inline org.learningequality.Kolibri.yaml:26]: How forwards- and backwards-compatible are plugins versus the core?

- Will Kolibri of today work with the Explore plugin of 12 months from now?
- Will the Explore plugin of today work with Kolibri of 12 months from now?

If no, this extension point should probably be versioned with e.g. the Kolibri major version or whatever the "plugin-api-stable" guarantee is. Flatpak will typically try to update apps' extensions at the same time as the app but this is not guaranteed.

Comment (manuq) [reply org.learningequality.Kolibri.yaml:26]: Indeed. The 2 versions are tightly coupled. The current kolibri-explore-plugin is known to work for Kolibri v0.14.6 only.

**Comment by dylanmccall** [comment]
This pull request is pretty old, so I created a new one with a newly rebased branch: #58

**Pull request closed without merging**